### PR TITLE
Record marsupial pairs

### PR DESCRIPTION
**Pull request by nkoenig**
Record marsupial pairs in the summary.yaml file.

### CODE REVIEW

File: subt_ign/src/GameLogicPlugin.cc [hunk 1]
```diff
@@ -640,6 +644,31 @@ void GameLogicPlugin::PostUpdate(
...
+
+          auto parentName = _ecm.Component<gazebo::components::Name>(
+              parentModel->Data());
+
+          auto childName = _ecm.Component<gazebo::components::Name>(
+              childModel->Data());
+          // Store the marsupial pairs, and handle the special case of
+          // "X1_platform"
+          if (parentName->Data() != "X1_platform" &&
+              childName->Data() != "X1_platform")
```
Comment (azeey) [inline subt_ign/src/GameLogicPlugin.cc:664]: The platform is actually going to be named `<name>_platform`, so maybe check if the name ends with `_platform`? If we filter out non-performers, we won't have to do this check.

File: subt_ign/src/GameLogicPlugin.cc [hunk 2]
```diff
@@ -640,6 +644,31 @@ void GameLogicPlugin::PostUpdate(
     std::map<std::string, ignition::math::Pose3d>::iterator baseIter =
       this->dataPtr->poses.find(subt::kBaseStationName);
 
+    _ecm.Each<gazebo::components::DetachableJoint>(
```
Comment (azeey) [inline subt_ign/src/GameLogicPlugin.cc:647]: We use detachable joints for making breadcrumbs and dynamic obstacles static after a period of time. We can maybe filter them out by checking if both the parent and child models are also performers.

File: subt_ign/src/GameLogicPlugin.cc (hunk 1 above)
Comment (nkoenig) [reply subt_ign/src/GameLogicPlugin.cc]: I think this fixes it: 9d1b038f198dab3f825d7780e6503e657076ab51

File: subt_ign/src/GameLogicPlugin.cc (hunk 2 above)
Comment (nkoenig) [reply subt_ign/src/GameLogicPlugin.cc:648]: I think this fixes it: 9d1b038f198dab3f825d7780e6503e657076ab51

File: subt_ign/src/GameLogicPlugin.cc [hunk 3]
```diff
@@ -640,6 +645,52 @@ void GameLogicPlugin::PostUpdate(
...
+          bool childPerformer = false;
+          for (const auto &t :
+              _ecm.ChildrenByComponents(childModel->Data(),
+                gazebo::components::Performer()))
+          {
+            childPerformer = true;
+            break;
+          }
+
+          auto parentName = _ecm.Component<gazebo::components::Name>(
```
Comment (azeey) [inline subt_ign/src/GameLogicPlugin.cc:675]: nit: `parentName` and `childName` are defined again below.

File: subt_ign/src/GameLogicPlugin.cc [hunk 4]
```diff
@@ -1561,6 +1612,9 @@ std::chrono::steady_clock::time_point GameLogicPluginPrivate::UpdateScoreFiles()
 
   // Output a run summary
   std::ofstream summary(this->logPath + "/summary.yml", std::ios::out);
+  summary << "marsupials:\n";
```
Comment (azeey) [inline subt_ign/src/GameLogicPlugin.cc:1615]: If there are no marsupials, this will just print `marsupials:`. Is that okay?

Comment (nkoenig) [reply subt_ign/src/GameLogicPlugin.cc:1615]: I think that is still valid yaml, and would indicate no marsupials.

File: subt_ign/src/GameLogicPlugin.cc (hunk 3 above)
Comment (nkoenig) [reply subt_ign/src/GameLogicPlugin.cc]: Fixed.

**Pull request merged**